tools/Customization/DevHome.Customization/ViewModels/DevDriveInsightsViewModel.cs

```diff
@@ -227,33 +232,60 @@ public void UpdateListViewModelList()
         DevDriveLoadingCompleted = true;
     }
 
-    private string? GetExistingCacheLocation(string rootDirectory, string targetDirectoryName)
-    {
-        var fullDirectoryPath = rootDirectory + targetDirectoryName;
-        if (Directory.Exists(fullDirectoryPath))
+    private readonly List<DevDriveCacheData> _cacheInfo =
+    [
+        new DevDriveCacheData
+        {
+            CacheName = "Pip cache (Python)",
+            EnvironmentVariable = "PIP_CACHE_DIR",
+            CacheDirectory = new List<string>
+            {
+                Path.Join(_localAppDataPath, "pip", "cache"),
+                Path.Join(_localAppDataPath, "packages", "PythonSoftwareFoundation.Python"),
+            },
+            ExampleDirectory = Path.Join("D:", "packages", "pip", "cache"),
+        },
+        new DevDriveCacheData
         {
-            return fullDirectoryPath;
+            CacheName = "NuGet cache (dotnet)",
+            EnvironmentVariable = "NUGET_PACKAGES",
+            CacheDirectory = new List<string> { Path.Join(_userProfilePath, ".nuget", "packages") },
+            ExampleDirectory = Path.Join("D:", "packages", "NuGet", "Cache"),
         }
-        else
+    ];
+
+    private string? GetExistingCacheLocation(DevDriveCacheData cache)
+    {
+        foreach (var cacheDirectory in cache.CacheDirectory!)
         {
-            var subDirPrefix = rootDirectory + "\\Packages\\PythonSoftwareFoundation.Python";
-            var subDirectories = Directory.GetDirectories(rootDirectory + "\\Packages", "*", SearchOption.TopDirectoryOnly);
-            var matchingSubdirectory = subDirectories.FirstOrDefault(subdir => subdir.StartsWith(subDirPrefix, StringComparison.OrdinalIgnoreCase));
-            var alternateFullDirectoryPath = matchingSubdirectory + "\\localcache\\local" + targetDirectoryName;
-            if (Directory.Exists(alternateFullDirectoryPath))
+            if (Directory.Exists(cacheDirectory))
             {
-                return alternateFullDirectoryPath;
+                return cacheDirectory;
+            }
+            else
+            {
+                var subDirectories = Directory.GetDirectories(_localAppDataPath + "\\Packages", "*", SearchOption.TopDirectoryOnly);
+                var matchingSubdirectory = subDirectories.FirstOrDefault(subdir => subdir.StartsWith(cacheDirectory, StringComparison.OrdinalIgnoreCase));
+                if (Directory.Exists(matchingSubdirectory))
+                {
+                    if (matchingSubdirectory.Contains("PythonSoftwareFoundation"))

```

Comment:
>if (matchingSubdirectory.Contains("PythonSoftwareFoundation")) [](http://example.com/codeflow?start=20&length=62)

Do we need this check? We are already searching for the same in the previous line.

Reply:
this is needed as there is specific logic for the PythonSoftwareFoundation path that the other packages won't have.

Especially since each python version has a different path altogether.

Reply:
You are right.